src/communication_interface.cc

```diff
@@ -218,8 +223,18 @@ void CommunicationInterface::PublishRobotStatus() {
     lock.unlock();
 
     lcm_.publish(params_.lcm_status_channel, &franka_status);
-    PublishBoolToChannel(franka_status.utime, lcm_user_stop_channel_,
-                         current_mode == franka::RobotMode::kUserStopped);
+
+    // publish true to user_stop_channel if U-stop is simulated
+    if(sim_u_stop){
+      PublishBoolToChannel(franka_status.utime, lcm_user_stop_channel_,
+                           true);
+    }
```

Comment:
Why not just set `current_mode = franka::RobotMode::kUserStopped` and avoid duplicate logic here? that way you can avoid needing the `sim_u_stop` variable completely